### PR TITLE
User challenge validation [MP-32]

### PR DESCRIPTION
**Pull request by BernardSimpson**
feat: Made slight changes to challenge.rb. addd maximum characters for both title and details.
feat: Created validation for pathway.rb. This includes title, details and difficulty.
feat: user_challenge.rb validation added.

### CODE REVIEW

**Comment by CraigNorford** [comment]
You have a conflict that needs resolving

File: app/models/user_challenge.rb [hunk 1]
```diff
@@ -1,2 +1,6 @@
 class UserChallenge < ApplicationRecord
-end
+    validates :completed, presence: true
+    validates :notes, presence: true, length: { maximum: 1500, }
```
Comment (CraigDoesCode) [inline app/models/user_challenge.rb:3]: I think a user challenge should be able to be created with out notes or feedback and both can be added at a later date. so we shouldn't validate presence true for those two fields

File: db/seeds.rb [hunk 2]
```diff
@@ -13,3 +14,28 @@
...
+# Challenge.create( completed: "28/10/2022", notes: "Just to advise you that this is a test run.", feedback: "You are doing really well, keep it up son", rating: 4, user_id: User.last.id)
+
+
+# This file should contain all the record creation needed to seed the database with its default values.
+# The data can then be loaded with the bin/rails db:seed command (or created alongside the database with db:setup).
+#
+# Examples:
+#
+#   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
+#   Character.create(name: "Luke", movie: movies.first)
```
Comment (CraigDoesCode) [inline db/seeds.rb:38]: Can we remove the commented out code here also

**Comment by BernardSimpson** [comment]
closed and creating new commit with changes

**Pull request closed without merging**